doc/examples/jupyter_notebook_prototyping/jupyter_notebook_prototyping_tutorial.rst

```diff
@@ -0,0 +1,294 @@
+
+        # set goal using a pose message this time
+        pose_goal = PoseStamped()
+        pose_goal.header.frame_id = "panda_link0"
+        pose_goal.pose.orientation.w = 1.0
+        pose_goal.pose.position.x = 0.28
+        pose_goal.pose.position.y = -0.2
+        pose_goal.pose.position.z = 0.5
+        panda_arm.set_goal(goal_pose_msg = pose_goal, link_name = "panda_link8")
+
+        # plan to goal
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                robot_trajectory = plan_result.trajectory
+                moveit.execute(robot_trajectory, blocking=True, controllers=[])
+
+Since we are using a notebook this mistake is easy to rectify without having to recompile any files. Simply edit the above notebook to match the below and rerun the cell: ::
+
+        # set plan start state using predefined state
+        panda_arm.set_start_state_to_current_state() # This conflicts with the current robot pose and will cause and error
+
+        # set goal using a pose message this time
+        pose_goal = PoseStamped()
+        pose_goal.header.frame_id = "panda_link0"
+        pose_goal.pose.orientation.w = 1.0
+        pose_goal.pose.position.x = 0.28
+        pose_goal.pose.position.y = -0.2
+        pose_goal.pose.position.z = 0.5
+        panda_arm.set_goal(goal_pose_msg = pose_goal, link_name = "panda_link8")
+
+        # plan to goal
+        plan_result = panda_arm.plan()
+
+        # execute the plan
+        if plan_result:
+                robot_trajectory = plan_result.trajectory
+                moveit.execute(robot_trajectory, blocking=True, controllers=[])
+
+Teleoperation Example
+---------------------
+
+One may also want to perform live teleoperation of their robot. With the Python API it is possible to interactively start/stop teleoperation without shutting down and subsequently relaunching all processes. In this example, we are going to show how this is possible with notebooks through a motivating example of teleoperating the robot, performing motion planning and teleoperating the robot again.
+
+For this section you will need a device that support teleoperation with ``moveit_py``, in this case we leverage the PS4 dualshock controller.
+
+We start by launching a servo client through directly launching the following `launch file <https://github.com/peterdavidfagan/moveit2_tutorials/blob/moveit_py_notebook_tutorial/doc/examples/jupyter_notebook_prototyping/launch/notebook_start_servo_client.launch.py>`_: ::
+
+        # launch servo client from within notebook
+        !ros2 launch moveit2_tutorials notebook_start_servo_client.launch.py
+
+We can then start the teleoperation node: ::
+
+        from moveit.servo_client.devices.ps4_dualshock import PS4DualShockTeleop
+
+        # instantiate the teleoperating device
+        ps4 = PS4DualShockTeleop()
+
+        # start teleloperating the robot
+        ps4.start_teleop()
```

Comment:
When running this, I got this error:

```
TypeError: PS4DualShockTeleop.__init__() missing 1 required positional argument: 'ee_frame_name'
```

As the error message says, this works, for instance

```
ps4 = PS4DualShockTeleop(ee_frame_name="panda_hand")
```

Reply:
I made a breaking change to the library which causes this error recently. I will update the notebook, ideally this should be pulled from parameters so I am adding an issue to update this.

Reply:
https://github.com/ros-planning/moveit2/issues/1969